R/git-utils.R

```diff
@@ -112,6 +112,11 @@ git_branch_exists <- function(branch) {
 }
 
 git_branch_tracking <- function(branch = git_branch_name()) {
+    b <- git_branch(name = branch)
+    git2r::branch_get_upstream(b)$name
+}
+
+git_branch_tracking_FIXME <- function(branch = git_branch_name()) {
```

Comment:
Expedient way to do as the commit says:

> Make `git_branch_tracking()` do what it says on the tin
>
> Not sure what to call the original version; it will temporarily have
a terrible name until I figure that out.

Reply:
`git_branch_tracking_FIXME()` is the function that is 50% "report tracking branch" and 50% "return what we think tracking branch should be".

It's not wrong, but it will need to be reworked. It was written to support `pr_*()` workflows, a very specific task.